crates/uv-fs/src/lib.rs

```diff
@@ -177,6 +179,24 @@ pub fn remove_symlink(path: impl AsRef<Path>) -> std::io::Result<()> {
     }
 }
 
+pub fn symlink_exists(path: impl AsRef<Path>) -> bool {
+    #[cfg(unix)]
+    {
+        path.as_ref()
+            .symlink_metadata()
+            .map(|metadata| metadata.file_type().is_symlink())
+            .unwrap_or(false)
+    }
+    #[cfg(windows)]
+    {
+        path.as_ref().symlink_metadata().is_ok_and(|metadata| {
+            // Check file attributes for the reparse point attribute
+            // FILE_ATTRIBUTE_REPARSE_POINT = 0x400
+            (metadata.file_attributes() & 0x400) != 0
+        })
```

Comment:
This might need a little more commentary, I'm not sure what this means or why it's here despite the comment.

Reply:
I've made it explicit that the reparse point attribute indicates this is a symlink or a junction.